### PR TITLE
fix(private-vpn): add privatek8s-tier subnet as interface

### PR DESCRIPTION
**Pull request by lemeurherve**

### CODE REVIEW

File: hieradata/clients/private.vpn.jenkins.io.yaml [hunk 1]
```diff
@@ -1,37 +1,18 @@
...
-    # # network_cidr: 10.0.1.0/24
-    # network_cidr: 10.244.0.0/14
+  eth1:
+    # Defined in https://github.com/jenkins-infra/azure/blob/main/privatek8s.tf
+    name: Subnet 'privatek8s-tier' of network 'private-vnet'
+    route-metric: 200
+    network_cidr: 10.249.0.0/16
+    peered_network_cidrs:
+      # Network 'private-vnet' through peering
+      - 10.248.0.0/14
```
Comment (dduportal) [inline hieradata/clients/private.vpn.jenkins.io.yaml]:
```suggestion
    # Defined in https://github.com/jenkins-infra/azure/blob/main/privatek8s.tf
    name: Network 'private-vnet'
    route-metric: 200
    network_cidr: 10.248.0.0/14      
```

After carefully reviwing https://github.com/lemeurherve/jenkins-infra/blob/c3eeb9d2328a4c0778b58fbdd1366c80cf9266c9/dist/profile/manifests/openvpn.pp#L127-L146, it appears that the `network_cidr` is only used for the routing table, not for the NIC configuration.

It means we can simplify by specifying the whole network containint the `eth1`'s NIC subnet

Comment (lemeurherve) [reply hieradata/clients/private.vpn.jenkins.io.yaml]:
```suggestion
    # Defined in https://github.com/jenkins-infra/azure-net/blob/main/vnet.tf
    name: 'private-vnet'
    route-metric: 200
    network_cidr: 10.248.0.0/14
```

**Review by dduportal — DISMISSED** [review]

File: hieradata/clients/private.vpn.jenkins.io.yaml [hunk 2]
```diff
@@ -1,37 +1,15 @@
 ---
 profile::openvpn::networks:
   eth0:
+    # Defined in https://github.com/jenkins-infra/azure-net/blob/main/vnets.tf
     name: Subnet 'app-tier' of network 'public-vnet'
```
Comment (lemeurherve) [inline hieradata/clients/private.vpn.jenkins.io.yaml]:
```suggestion
    name: 'public-vnet'
```

File: hieradata/clients/private.vpn.jenkins.io.yaml [hunk 3]
```diff
@@ -1,37 +1,15 @@
...
     route-metric: 100
-    # macaddress: 00:0d:3a:0e:4e:7e
-    # network_cidr: 10.0.1.0/24
     network_cidr: 10.244.0.0/14
```
Comment (dduportal) [inline hieradata/clients/private.vpn.jenkins.io.yaml]:
```suggestion
    network_cidr: 10.248.0.0/28
```

as per https://github.com/jenkins-infra/azure-net/blob/96a2f2d4172677575624a7bcb6e96a2ca41b6cef/vnets.tf#L50

**Review by dduportal — APPROVED** [review]

**Pull request merged**